packages/react-native-executorch/common/rnexecutorch/models/ocr/RecognitionHandler.cpp

```diff
@@ -0,0 +1,81 @@
+#include "RecognitionHandler.h"
+#include "RecognitionHandlerUtils.h"
+#include <rnexecutorch/Log.h>
+#include <rnexecutorch/data_processing/ImageProcessing.h>
+#include <rnexecutorch/models/ocr/Constants.h>
+#include <rnexecutorch/models/ocr/RecognitionHandlerUtils.h>
+
+namespace rnexecutorch {
+RecognitionHandler::RecognitionHandler(
+    std::string recognizerSourceLarge, std::string recognizerSourceMedium,
+    std::string recognizerSourceSmall, std::string symbols,
+    std::shared_ptr<react::CallInvoker> callInvoker)
+    : converter(symbols), recognizerLarge(recognizerSourceLarge, callInvoker),
+      recognizerMedium(recognizerSourceMedium, callInvoker),
+      recognizerSmall(recognizerSourceSmall, callInvoker) {
+  memorySizeLowerBound = recognizerSmall.getMemoryLowerBound() +
+                         recognizerMedium.getMemoryLowerBound() +
+                         recognizerLarge.getMemoryLowerBound();
+}
+
+std::pair<std::vector<int32_t>, float>
+RecognitionHandler::runModel(cv::Mat image) {
+  if (image.cols >= ocr::largeRecognizerWidth) {
+    return recognizerLarge.generate(image);
+  } else if (image.cols >= ocr::mediumRecognizerWidth) {
```

Comment:
Previous branch returned, so you can use if instead of else if, there is a clang tidy check for this one ;)
```suggestion
  if (image.cols >= ocr::mediumRecognizerWidth) {
```